dist/ethereum.js

```diff
@@ -2604,39 +2615,65 @@ var requestManager = function() {
         }
 
         // HTTP ASYNC (only when callback is given, and it a HttpProvidor)
-        if(typeof callback === 'function' && provider.name === 'HTTP'){
+        if(typeof callback === 'function' && provider.name === 'RPC over HTTP'){
```

Comment:
Checking provider name and creating provider specific logic here is against all good practices. Let's assume, that we have 10 async providers. Would you create `if` for each of them?